src/java/picard/vcf/GenotypeConcordanceCounts.java

```diff
@@ -74,6 +85,76 @@ public void validateCountsAgainstScheme(final GenotypeConcordanceScheme scheme)
     }
 
     /**
+     * Genotype Concordance is the number of times the truth and call states match exactly / all truth and call combinations made
+     * If the GA4GH scheme is being used, any MISSING sites in truth OR call will not be included in the discordance calculations.
+     */
+    public double calculateGenotypeConcordance(final GenotypeConcordanceScheme scheme, final boolean missingSitesFlag) {
+        double numerator = 0.0;
+        double denominator = 0.0;
+
+        scheme.validateScheme();
+
+        final TruthState[] allTruthStates = TruthState.values();
+        final CallState[] allCallStates = CallState.values();
+
+        for (final TruthState truthState : allTruthStates) {
+            for (final CallState callState : allCallStates) {
+                if(!missingSitesFlag && isMissing(truthState, callState)) {
+                    continue;
+                }
+                else {
+                    final TruthAndCallStates truthAndCallStates = new TruthAndCallStates(truthState, callState);
+                    final int count = getCount(truthAndCallStates);
+                    if (truthState.getCode()==callState.getCode()) {
+                        //Because the TruthState and CallState enums are in the same order, the ordinals correspond to the same constants
+                        //If we enter this, we are 'on the diagonal
+                        numerator += count;
+                        denominator += count;
+                    } else {
+                        denominator += count;
+                    }
+                }
+            }
+        }
+        return (numerator / denominator);
+    }
+
+    /**
+     * Non Ref Genotype Concordance is the number of times the truth and call states match exactly for *vars only* / all truth and call *var* combinations made
+     * If the GA4GH scheme is being used, any MISSING sites in truth OR call will not be included in the discordance calculations.
+     */
+    public double calculateNonRefGenotypeConcordance(final GenotypeConcordanceScheme scheme, final boolean missingSitesFlag) {
+        double numerator = 0.0;
+        double denominator = 0.0;
+
+        scheme.validateScheme();
+
+        final TruthState[] allTruthStates = TruthState.values();
+        final CallState[] allCallStates = CallState.values();
+
+        for (final TruthState truthState : allTruthStates) {
+            for (final CallState callState : allCallStates) {
+                if(!missingSitesFlag && isMissing(truthState, callState)) {
+                    continue;
+                }
+                else if (isVar(truthState, callState)) {
```

Comment:
@kbergin @eitanbanks I agree with Eric's point here.  Given that both are calculating concordance it makes sense to have a private helper method.  If one was sensitivity and the other specificity, there would be more value in keeping the separate to be explicit.  It's a judgement call.